openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1952,3 +1952,28 @@ def import_blocks_create_task(library_key, course_key, use_course_key_as_block_i
     log.info(f"Import block task created: import_task={import_task} "
              f"celery_task={result.id}")
     return import_task
+
+
+def create_or_update_xblock_upstream_link(xblock, course_key: str, course_name: str, created: datetime | None = None):
+    """
+    Create or update upstream->downstream link in database for given xblock.
+    """
+    if not xblock.upstream:
+        log.info(f"No upstream found for xblock: {xblock.usage_key}")
+        return None
+    upstream_usage_key = UsageKeyV2.from_string(xblock.upstream)
+    try:
+        lib_component = get_component_from_usage_key(upstream_usage_key)
+    except ObjectDoesNotExist:
+        log.exception("Library block not found!")
```

Comment:
I don't think we need a `log.exception` here, because this could happen as part of normal operation, e.g if the lib component gets deleted. But it would be good to know a bit more about what it means:
```suggestion
        log.error(f"Library component not found for {upstream_usage_key} -- removing from upstream link")
```